### PR TITLE
[WIP] Support for multiple architectures

### PR DESCRIPTION
**Pull request by vuntz**
Support for multiple architectures was added there.

Depends on:
* https://github.com/crowbar/crowbar/pull/2091
* https://github.com/crowbar/barclamp-provisioner/pull/407
* https://github.com/SUSE-Cloud/suse-cloud-upgrade/pull/54
* https://github.com/cloudbase/adk-tools/pull/9
* [crowbar packaging change](http://paste.opensuse.org/view/raw/12452794)
* [barclamp-provisioner packaging change](http://paste.opensuse.org/view/raw/70497300)

### CODE REVIEW

File: crowbar_framework/app/helpers/application_helper.rb [hunk 1]
```diff
@@ -246,11 +246,11 @@ def default_platform
...
 
-  def available_platforms
-    NodeObject.available_platforms
+  def available_platforms(architecture)
+    NodeObject.available_platforms(architecture)
   end
 
-  def disabled_platforms
-    NodeObject.disabled_platforms
+  def disabled_platforms(architecture)(architecture)
```
Comment (bkutil) [inline crowbar_framework/app/helpers/application_helper.rb]: typo?

File: crowbar_framework/app/helpers/application_helper.rb [hunk 2]
```diff
@@ -246,11 +246,11 @@ def default_platform
     NodeObject.default_platform
   end
 
-  def available_platforms
-    NodeObject.available_platforms
+  def available_platforms(architecture)
```
Comment (bkutil) [inline crowbar_framework/app/helpers/application_helper.rb]: The callers have been modified, right?

**Comment by vuntz** [comment]
Still missing: default_platform needs to know about the architecture.

File: crowbar_framework/app/helpers/application_helper.rb (hunk 1 above)
Comment (vuntz) [inline crowbar_framework/app/helpers/application_helper.rb]: Fixed; caused by some annoying bug in `screen`/`vim` interaction :/

File: crowbar_framework/app/helpers/application_helper.rb (hunk 2 above)
Comment (vuntz) [inline crowbar_framework/app/helpers/application_helper.rb]: Yes

**Comment by tboerger** [comment]
LGTM

**Comment by rsalevsky** [comment]
@vuntz, this repository got merged into [crowbar-core](https://github.com/crowbar/crowbar-core).
Please open a new pull request there. __Do NOT merge here anymore!__

**Comment by vuntz** [comment]
Closing as this will need to be rebased on top of https://github.com/crowbar/crowbar-core/pull/18

**Pull request closed without merging**